autoload/db/adapter/bigquery.vim

```diff
@@ -0,0 +1,81 @@
+if !executable('bq')
+  echom 'bq command is not available.'
+  let g:autoloaded_db_bigquery = 0
+endif
+
+if exists('g:autoloaded_db_bigquery')
+  finish
+endif
+
+let g:autoloaded_db_bigquery = 1
+
+function! s:command_for_url(params, action) abort
+  let cmd = ['bq']
+  let subcmd = [a:action]
+
+  for [k, v] in items(a:params)
+    let op = ' --'.k.'='.v
+
+    if len(matchstr(k, 'legacy'))
+        " FIXME: parse subcmd options
+        call add(subcmd, op)
+    else
+        " parse global options
+        call add(cmd, op)
+    endif
+
+  endfor
+
+  return join(cmd + subcmd)
+
+endfunction
+
+function! s:params(url) abort
+  let parsed_params = db#url#parse(a:url)
+  let conn_params = parsed_params.params
+  return conn_params
+endfunction
+
+function! db#adapter#bigquery#interactive(url, action) abort
+  echom "Starting bigquery"
+  return split(s:command_for_url(s:params(a:url), a:action))
+endfunction
+
+function! db#adapter#bigquery#input(url, in) abort
+
+  " QST: what is the purpose of input?
```

Comment:
The purpose is to give the filename directly to the database, so that error messages, etc, can reference. Five adapters use this; the others accept input on stdin. Does bq really support neither?

Reply:
yes, it support it but I've faced some issue when I've tried to use it. I will review it.